test/system/helpers.bash

```diff
@@ -288,7 +288,7 @@ function wait_for_port() {
 
     # Wait
     while [ $_timeout -gt 0 ]; do
-        { exec 3<> /dev/tcp/$host/$port; } &>/dev/null && return
+        { exec 5<> /dev/tcp/$host/$port; } &>/dev/null && return
```

Comment:
@edsantiago you can avoid hardcoded file descriptors in bash entirely by using `{name}` feature, e.g.

```bash
exec {conn_fd} <> /dev/tcp/$host/$port
```
and then you can use `$conn_fd` variable which is a numeric FD that bash chose.

See https://github.com/opencontainers/runc/blob/master/tests/integration/checkpoint.bats for some more examples of this style (which I personally call "brackets and braces programming" 😄)

Reply:
@kolyshkin that is wonderful, I'm so glad to learn that! Thank you! (For future reference, should anyone find this thread: this feature is documented in `bash(1)` under `REDIRECTION`).